src/lib/components/submit/SubmitAddNodeProvider.svelte

```diff
@@ -90,6 +91,61 @@
 
 			debounceSave();
 		})();
+
+	export function checkFields(): boolean {
+		const validUrlDomain = 'https://wiki.internetcomputer.org/';
+		const sha256Regex = /^[a-fA-F0-9]{64}$/;
+		const fields = [
+			nodeProviderName,
+			url,
+			urlSelfDeclaration,
+			hashSelfDeclaration,
+			urlIdentityProof,
+			hashIdentityProof,
+			nodeProviderId
+		];
+
+		if (fields.some((field) => field === '')) {
+			toasts.error({
+				msg: { text: 'Please fill in all fields' }
+			});
+			return false;
+		}
+
+		if (!urlSelfDeclaration.startsWith(validUrlDomain)) {
```

Comment:
it would probably be cleaner to build an URL wdyt?

In case of exception, you already have a hint that it isn't a valid URL. Then you can compare information that are relevant, like `domain === expectedDomain` etc.